TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/EditCase/Reject.cshtml.cs

```diff
@@ -1,13 +1,93 @@
+using System.ComponentModel.DataAnnotations;
 using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeachingRecordSystem.Core.Dqt.Models;
+using TeachingRecordSystem.Core.Dqt.Queries;
 using TeachingRecordSystem.SupportUi.Infrastructure.Security;
 
 namespace TeachingRecordSystem.SupportUi.Pages.Cases.EditCase;
 
 [Authorize(Policy = AuthorizationPolicies.CaseManagement)]
 public class RejectModel : PageModel
 {
-    public void OnGet()
+    private readonly TrsLinkGenerator _linkGenerator;
+    private readonly ICrmQueryDispatcher _crmQueryDispatcher;
+
+    public RejectModel(
+        TrsLinkGenerator linkGenerator,
+        ICrmQueryDispatcher crmQueryDispatcher)
+    {
+        _linkGenerator = linkGenerator;
+        _crmQueryDispatcher = crmQueryDispatcher;
+    }
+
+    [FromRoute]
+    public string TicketNumber { get; set; } = null!;
+
+    [BindProperty]
+    [Display(Name = " ")]
+    [Required(ErrorMessage = "Select the reason for rejecting this change")]
+    public CaseRejectionReasonOption? RejectionReasonChoice { get; set; }
+
+    public async Task<IActionResult> OnGet()
+    {
+        (Incident Incident, dfeta_document[] Documents)? incidentAndDocuments = await GetIncidentAndDocuments();
+        if (incidentAndDocuments is null)
+        {
+            return NotFound();
+        }
+
+        if (incidentAndDocuments.Value.Incident.StateCode != IncidentState.Active)
+        {
+            return BadRequest();
+        }
+
+        return Page();
+    }
+
+    public async Task<IActionResult> OnPost()
+    {
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
+
+        (Incident Incident, dfeta_document[] Documents)? incidentAndDocuments = await GetIncidentAndDocuments();
```

Comment:
The alternative to re-getting the incident in order to get the incident ID is probably to refactor the various pages to use incident ID in the URL etc so it is passed in the route .. sort of like the ticket number in the URL but I guess it doesn't really matter if it was the incident ID instead